pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -142,6 +141,47 @@ func (cb *ClusterBuilder) applyDestinationRule(c *cluster.Cluster, clusterMode C
 	return subsetClusters
 }
 
+// MergeTrafficPolicy returns the merged TrafficPolicy for a destination-level and subset-level policy on a given port.
+func MergeTrafficPolicy(original, subsetPolicy *networking.TrafficPolicy, port *model.Port) *networking.TrafficPolicy {
+	if subsetPolicy == nil {
+		return original
+	}
+
+	mergedPolicy := original.DeepCopy()
```

Comment:
Do we need deepcopy here and below or can we just create a new mergerd `&networking.TrafficPolicy{}` and keep assigning the right settings?

Reply:
@ramaraochavali the issue that I ran into is that all of the settings that we want to assign in this function are pointers.  And in the case of the TLS context, in cluster.go we modify the TLS object during the upgrade to automtls or setting additional fields related to ISTIO_MUTUAL.  Using the unit test I modified in `cluster_test.go` as an example, there is a subset port level traffic policy for port 8080, but nothing for the other port 9090.  So when TLS settings were modified for 8080 during the merge, it was modifying the DestinationRule stored in push_context and it impacted subset 9090 as well.

TLS was the only object in TrafficPolicy that Istio modifies (that I am aware of) - but I figured a DeepCopy at the top level object would prevent something like this from leaking through again for the other settings

Reply:
Sure. Thanks. That makes sense

Reply:
Can you please add a comment there on why we are using DeepCopy?

Reply:
done